### PR TITLE
Update submodules

### PR DESCRIPTION
**Pull request by travitch**
This includes:

- Adapting for Crux API changes
- The split of what4 from crucible
- The bitvector-sized changes to what4/crucible

I'm doing this in preparation for another crux change I'd like to land.  I tried to run the tests but need some help with mir-json

### CODE REVIEW

**Comment by spernsteiner** [comment]
What trouble did you run into with mir-json?

File: src/Mir/Trans.hs [hunk 1]
```diff
@@ -687,11 +688,11 @@ evalCast' ck ty1 e ty2  =
       (M.Misc, TyBool, TyUint bsz)
        | MirExp C.BoolRepr e0 <- e
        -> baseSizeToNatCont bsz $ \w -> 
-           return $ MirExp (C.BVRepr w) (R.App $ E.BVIte e0 w (R.App $ E.BVLit w 1) (R.App $ E.BVLit w 0))
+           return $ MirExp (C.BVRepr w) (R.App $ E.BVIte e0 w (R.App $ E.BVLit w (BV.mkBV w 1)) (R.App $ E.BVLit w (BV.mkBV w 0)))
```
Comment (spernsteiner) [inline src/Mir/Trans.hs:691]: This pattern `BVLit w (mkBV w x)` shows up a lot here - would it make sense to define a common helper function?  I assume the same pattern would show up fairly often in other Crucible client code as well.

**Comment by travitch** [comment]
I've seen the same errors that the travis build reports.  I did eventually get a build of it, but the mir-verifier tests all fail with:
```
mir-verifier
  crux concrete

      box
        new:               FAIL (0.16s)
          Compiling and running oracle program (0.15s)
          Oracle output: ()                    (0.02s)
            user error (Error 101 while running mir-json on test/conc_eval/box/new.rs)
```

When I try to run `mir-json` manually, I get:
```
error[E0463]: can't find crate for `core`
```

File: src/Mir/Trans.hs (hunk 1 above)
Comment (travitch) [reply src/Mir/Trans.hs:691]: Yes, we could certainly introduce a small combinator to wrap that up.  We didn't do it in the other codebases to make the diffs as direct as possible for easier inspection.  I can introduce one here.

**Comment by spernsteiner** [comment]
>I've seen the same errors that the travis build reports

Looks like Travis is running into https://github.com/GaloisInc/mir-json/issues/17 - I'll work on a fix for that

>When I try to run mir-json manually, I get:
>error[E0463]: can't find crate for `core`

Did you run `./translate_libs.sh` first?  I see the same error when trying `cabal new-run  -- crux-mir test/conc_eval/box/new.rs -d3` after removing the translated libraries.

**Comment by travitch** [comment]
Oh, I didn't see that step.  Let me give that a shot - thanks!

**Pull request closed without merging**